lib/hanami/cli/generators/app/operation.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "erb"
+require "dry/files"
+require_relative "../../errors"
+
+module Hanami
+  module CLI
+    module Generators
+      module App
+        # @since x.x.x
+        # @api private
+        class Operation
+          # @since x.x.x
+          # @api private
+          def initialize(fs:, inflector:)
+            @fs = fs
+            @inflector = inflector
+          end
+
+          # @since x.x.x
+          # @api private
+          def call(app, key, slice)
+            context = OperationContext.new(inflector, app, slice, key)
+
+            if slice
+              generate_for_slice(context, slice)
+            else
+              generate_for_app(context)
+            end
+          end
+
+          private
+
+          attr_reader :fs
+
+          attr_reader :inflector
+
+          def generate_for_slice(context, slice)
+            slice_directory = fs.join("slices", slice)
+            raise MissingSliceError.new(slice) unless fs.directory?(slice_directory)
+
+            if context.namespaces.any?
+              fs.mkdir(directory = fs.join(slice_directory, context.namespaces))
+              fs.write(fs.join(directory, "#{context.name}.rb"), t("nested_slice_operation.erb", context))
+            else
+              fs.mkdir(directory = fs.join(slice_directory))
+              fs.write(fs.join(directory, "#{context.name}.rb"), t("top_level_slice_operation.erb", context))
+              fs.recommend("Add a namespace to operation names, so they go into a folder within #{directory}/.")
```

Comment:
What do you think about this for wording?

```suggestion
              fs.recommend(%(Generating a top-level operation. To generate into a directory, add a namespace: "my_namespace.#{context.name}))
```

That first sentence is there to explain why we're offering the recommendation in the first place. The second sentence is there to give a bit more guidance as to what the "recommended" args might look like.

Reply:
Sweet, switched it to this. I added the indentation too so it's more noticeable.